src/commands/publish/node/node-publish.command.ts

```diff
@@ -0,0 +1,44 @@
+import * as _path from 'path';
+import { Solution } from '../../../models/solution';
+import { ICommand } from '../../command';
+import { FileService } from '../../../services/file.service';
+import { DependencyTree } from '../../../models/dependency-tree';
+import { NodeEnvironmentService } from '../../../services/environments/node-environment.service';
+import { IEnvironmentService } from '../../../services/environments/environment.service';
+
+export class NodePublishCommand implements ICommand {
+    get name(): string { return "publish-node" };
+    fileService = new FileService();
+    environmentService: IEnvironmentService = new NodeEnvironmentService();
+
+    run = (solutionFilePath: string): Promise<void> => {
+        solutionFilePath = _path.join(process.cwd(), solutionFilePath);
+        return this.fileService.getShamanFile(solutionFilePath)
+            .then(solution => this.publishSolution(solutionFilePath, solution));
+    }
+
+    private publishSolution = (solutionFilePath: string, solution: Solution): Promise<void> => {
+        let cwd = solutionFilePath.replace('shaman.json', '');
+        let binPath = _path.join(cwd, 'bin/');
+        let projects = solution.projects.filter(p => p.environment == "node");
+        if (!projects.length) return Promise.resolve();
+        console.log("Publishing node projects...");
+        let dependencyTree = new DependencyTree(projects);
+        let publishOrder = dependencyTree.getOrderedProjectList();
+        return this.createEnvironmentPublishFolder(binPath)
+            .then(_ => publishOrder.reduce((a, b) => a.then(_ => {
+                let project = solution.projects.find(p => p.name == b);
+                let destinationPath = _path.join(binPath, "node", project.path);
+                return this.environmentService.publishProject(project.name, _path.join(cwd, project.path), destinationPath)
```

Comment:
I can tell just by looking at this that something is missing: the build step. After looking deeper, I saw that inside the node environment service's publish method, but that is an indication that something is fishy (I think this is probably what you were talking about in your text message). Please see comment in the node environment service.